src/components/Search/SearchRouter/SearchRouterList.tsx

```diff
@@ -351,14 +357,59 @@ function SearchRouterList(
         cardAutocompleteList,
     ]);
 
-    const sections: Array<SectionListDataType<OptionData | SearchQueryItem>> = [];
+    const sortedRecentSearches = useMemo(() => {
+        return Object.values(recentSearches ?? {}).sort((a, b) => b.timestamp.localeCompare(a.timestamp));
+    }, [recentSearches]);
+
+    const recentSearchesData = sortedRecentSearches?.slice(0, 5).map(({query, timestamp}) => {
+        const searchQueryJSON = SearchQueryUtils.buildSearchQueryJSON(query);
+        return {
+            text: searchQueryJSON ? SearchQueryUtils.buildUserReadableQueryString(searchQueryJSON, personalDetails, reports, taxRates) : query,
+            singleIcon: Expensicons.History,
+            searchQuery: query,
+            keyForList: timestamp,
+            searchItemType: CONST.SEARCH.SEARCH_ROUTER_ITEM_TYPE.SEARCH,
+        };
+    });
+
+    const recentReportsOptions = useMemo(() => {
+        if (autocompleteQueryValue.trim() === '') {
+            return searchOptions.recentReports.slice(0, 20);
+        }
 
-    const [isInitialRender, setIsInitialRender] = useState(true);
+        Timing.start(CONST.TIMING.SEARCH_FILTER_OPTIONS);
+        const filteredOptions = OptionsListUtils.filterOptions(searchOptions, autocompleteQueryValue, {sortByReportTypeInSearch: true, preferChatroomsOverThreads: true});
+        Timing.end(CONST.TIMING.SEARCH_FILTER_OPTIONS);
+
+        const reportOptions: OptionData[] = [...filteredOptions.recentReports, ...filteredOptions.personalDetails];
+        if (filteredOptions.userToInvite) {
+            reportOptions.push(filteredOptions.userToInvite);
+        }
+        return reportOptions.slice(0, 20);
+    }, [autocompleteQueryValue, searchOptions]);
+
+    useEffect(() => {
+        ReportUserActions.searchInServer(autocompleteQueryValue.trim());
+    }, [autocompleteQueryValue]);
```

Comment:
The `autocompleteQueryValue` contains filters that shouldn't be passed to the BE report search endpoint as they are irrelevant and causes two identical search values to be treated differently resulting in 2 API calls e.g.

- `type:expense status:all group-by:reports ExpenseReportName`
- `type:expense status:drafts group-by:reports ExpenseReportName`

In both queries the report that we are searching for is `ExpenseReportName`. That's all what we should send to the server and only do it once.

Coming from https://github.com/Expensify/App/issues/60336